### PR TITLE
Prune carnot old block

### PR DESCRIPTION
**Pull request by danielSanchezQ**

### CODE REVIEW

**Comment by bacv** [comment]
For this to work with `cli` app, we need to move `carnot/blocks` endpoint to `storage/blocks`, the app at the moment pull latest 500 blocks directly from consensus.

**Comment by danielSanchezQ** [comment]
> For this to work with `cli` app, we need to move `carnot/blocks` endpoint to `storage/blocks`, the app at the moment pull latest 500 blocks directly from consensus.

Yes, I was aware. There is a block explorer PR that supplies that. But probably needs some updates. That is why I open this as a draft.

**Comment by bacv** [comment]
I think block explorer is really worth continuing, when it's ready, this change could be done.

**Comment by danielSanchezQ** [comment]
@bacv do you think we could use of https://github.com/logos-co/nomos-node/pull/558 better for now? Until the block explorer is ready an integrated?

**Comment by bacv** [comment]
https://github.com/logos-co/nomos-node/pull/558 would allow the cli app to work (pruning controlled by the client). Even though it's not the final version, it would allow us to use testnet from the master branch (if merged). 

On the other hand, we won't need api provided by #558 if we have block explorer and pruning at the consensus level, so it feels bit redundant.

**Comment by zeegomo** [comment]
IIRC, there was some discussion about avoiding complicated apis in the node, but we need a way to provide blocks.

Regardless of that, we should not keep block headers forever and this PR does the right thing. Just make sure it does not break the chat app when we merge it

File: nomos-services/carnot-consensus/src/lib.rs [hunk 1]
```diff
@@ -642,6 +642,10 @@ where
                             committee_membership.on_new_block_received(&original_block)
                         },
                     );
+                    // prune old blocks, older than grand parent
+                    new_state.prune_older_blocks_by_view(
+                        new_state.latest_committed_view().prev().prev(),
+                    );
```
Comment (zeegomo) [inline nomos-services/carnot-consensus/src/lib.rs:648]: nit: technically we can remove anything that is before the latest commited block (no need to get its grandparent)

**Review by zeegomo — APPROVED** [review]

File: nomos-services/carnot-consensus/src/lib.rs (hunk 1 above)
Comment (zeegomo) [reply nomos-services/carnot-consensus/src/lib.rs:648]: yes, we can even keep the last X blocks if that's helpful

**Review by al8n — APPROVED** [review]

**Comment by danielSanchezQ** [comment]
Closing as we removed Carnot

**Pull request closed without merging**